### PR TITLE
Fix bug in check table for tables without primary key

### LINKED ISSUE #7543: CHECK TABLE (and ReplicatedMergeTreePartCheckThread) always fails when primary key is empty.
```
:) create table stupid_mt (SomeField1 Int64, SomeField2 Double) Engine =MergeTree() Order by tuple();

insert into stupid_mt  VALUES (1, 2);

check table stupid_mt;


┌─part_path─┬─is_passed─┬─message───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ all_1_1_0 │         0 │ Cannot open file /home/alesap/clickdb/data/default/stupid_mt/all_1_1_0/primary.idx, errno: 2, strerror: No such file or directory │
└───────────┴───────────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (up to few sentences, required except for Non-significant/Documentation categories):
Fix `CHECK TABLE` query for `*MergeTree` tables without key. Fixes [#7543](https://github.com/ClickHouse/ClickHouse/issues/7543).

### CODE REVIEW

**Review by tavplubix — APPROVED** [review]

File: dbms/src/Storages/MergeTree/checkDataPart.cpp [hunk 1]
```diff
@@ -217,6 +217,7 @@ MergeTreeData::DataPart::Checksums checkDataPart(
     MergeTreeData::DataPart::Checksums checksums_data;
 
     size_t marks_in_primary_key = 0;
+    if (Poco::File(path + "primary.idx").exists())
```
Comment (alexey-milovidov) [inline dbms/src/Storages/MergeTree/checkDataPart.cpp]: Looks like incorrect change. What if primary key must exist but the file was removed.

Why don't just move `if (!primary_key_data_types.empty())` upper?

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

File: dbms/src/Storages/MergeTree/checkDataPart.cpp (hunk 1 above)
Comment (alexey-milovidov) [reply dbms/src/Storages/MergeTree/checkDataPart.cpp]: Ok.
`primary_key_data_types` can be empty to skip check of the primary key.
But it's needed only for `part_checker` tool. I'd better remove `part_checker` tool.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by nikitamikhaylov** [comment]
Conflicts with 19.17

**Pull request merged**